### DISCUSSION TITLE
kitten notify not working on macOS Sonoma

### DISCUSSION THREAD

**Discussion by sinkr** [kovidgoyal/kitty#7929] [Q&A] [answered]
Hello,

Using the standard example in v0.36.4:
```
kitten notify --wait-for-completion --button One --button Two "Good morning" Hello world, it is a nice day!
```

Exits immediately and does not display a notification, however:
```
osascript -e 'display notification "TEST" with title "Kitty"'
```

Does return a notification.

I don't see Kitty as an option in Settings -> Notifications Settings, which I feel is the issue.

Any ideas?

**Comment by kovidgoyal** [comment, marked as answer]
Works fine on my sonoma machine, and there is an entry for kitty in
Settings -> Notifications. Make sure you are using the official signed
kitty binaries. Apple doesnt allow unsigned applications to post notifications.

**Reply by sinkr** [reply]
> (...) Apple doesnt allow unsigned applications to post notifications.

Ah, that would be it!  Self signing corrected the problem--thank you for the response and great app!

**Reply by gachikuku** [reply]
How do I self sign kitten. 

I done `codesign -s my-codesign-cert /opt/homebrew/bin/kitten`(from [StackOverflow](https://stackoverflow.com/questions/58356844/what-are-the-ways-or-technologies-to-sign-an-executable-application-file-in-mac))
and still I was not being able to get a notification using kitten.

**Reply by gachikuku** [reply]
Never mind, I was inside `tmux` that's why nothing was working.

**Comment by postsolar** [comment]
@kovidgoyal could you please clarify if ad-hoc signing is meant to work? This is what nixpkgs uses, but for some reason in this kitty notifications don't work, whereas they do in kitty installed via the official install script. I'm on Tahoe 26.3 Beta.

**Reply by kovidgoyal** [reply]
No idea, you'd have to ask apple. Since I am forced to pay Apple's protection racket money every year I havent used ad hoc signing in many many years.
